packages/@tailwindcss-upgrade/src/template/migrate.ts

```diff
@@ -0,0 +1,45 @@
+import fs from 'node:fs/promises'
+import path from 'node:path'
+import type { Candidate } from '../../../tailwindcss/src/candidate'
+import type { DesignSystem } from '../../../tailwindcss/src/design-system'
+import { extractCandidates, printCandidate, replaceCandidateInContent } from './candidates'
+import { migrateImportant } from './codemods/migrate-important'
+
+export type Migration = (candidate: Candidate) => Candidate | null
+
+export default async function migrateContents(
+  designSystem: DesignSystem,
+  contents: string,
+  migrations: Migration[] = [migrateImportant],
+): Promise<string> {
+  let candidates = await extractCandidates(designSystem, contents)
+
+  // Sort candidates by starting position desc
+  candidates.sort((a, z) => z.start - a.start)
+
+  let output = contents
+  for (let { candidate, start, end } of candidates) {
+    let needsMigration = false
+    for (let migration of migrations) {
+      let migrated = migration(candidate)
+      if (migrated) {
+        candidate = migrated
+        needsMigration = true
+        break
```

Comment:
Is this `break` correct? If you have 2 migrations, and the first migration results in "Yep, let's print the new candidate" then the second migration won't run.

Reply:
We only have a single migration so doesn't matter right now haha